zebra-network/src/peer/client/tests.rs

```diff
@@ -1,3 +1,153 @@
+            version,
+        };
+
+        let handle = MockedClientHandle {
+            request_receiver: Some(request_receiver),
+            shutdown_receiver: Some(shutdown_receiver),
+            error_slot,
+            version,
+        };
+
+        (handle, client)
+    }
+
+    /// Gets the peer protocol version associated to the [`Client`].
+    pub fn version(&self) -> Version {
+        self.version
+    }
+
+    /// Checks if the [`Client`] instance has not been dropped, which would have disconnected from
+    /// the peer.
+    pub fn is_connected(&mut self) -> bool {
+        let receive_result = self
+            .shutdown_receiver
+            .as_mut()
+            .expect("shutdown receiver has been dropped")
+            .try_recv();
+
+        match receive_result {
+            Ok(None) => true,
+            Ok(Some(CancelHeartbeatTask)) | Err(oneshot::Canceled) => false,
+        }
+    }
+
+    /// Drops the shutdown receiver endpoint.
+    pub fn drop_shutdown_receiver(&mut self) {
+        let _ = self
+            .shutdown_receiver
+            .take()
+            .expect("request receiver endpoint has already been dropped");
+    }
+
+    /// Closes the request receiver endpoint.
+    pub fn close_request_receiver(&mut self) {
+        self.request_receiver
+            .as_mut()
+            .expect("request receiver endpoint has been dropped")
+            .close();
+    }
+
+    /// Drops the request receiver endpoint, forcefully closing the channel.
+    pub fn drop_request_receiver(&mut self) {
+        self.request_receiver
+            .take()
+            .expect("request receiver endpoint has already been dropped");
+    }
+
+    /// Tries to receive a [`ClientRequest`] sent by the mocked [`Client`] instance.
+    pub(crate) fn try_to_receive_request(&mut self) -> ReceiveRequestAttempt {
+        let receive_result = self
+            .request_receiver
+            .as_mut()
+            .expect("request receiver endpoint has been dropped")
+            .try_next();
+
+        match receive_result {
+            Ok(Some(request)) => ReceiveRequestAttempt::Request(request),
+            Ok(None) => ReceiveRequestAttempt::Closed,
+            Err(_) => ReceiveRequestAttempt::Empty,
+        }
+    }
+
+    /// Returns the current error in the [`ErrorSlot`], if there is one.
+    pub fn current_error(&self) -> Option<SharedPeerError> {
+        self.error_slot.try_get_error()
+    }
+
+    /// Sets the error in the [`ErrorSlot`], assuming there isn't one already.
+    ///
+    /// # Panics
+    ///
+    /// If there's already an error in the [`ErrorSlot`].
+    pub fn set_error(&self, error: impl Into<SharedPeerError>) {
+        self.error_slot
+            .try_update_error(error.into())
+            .expect("unexpected earlier error in error slot")
+    }
+}
+
+/// A representation of the result of an attempt to receive a [`ClientRequest`] sent by the mocked
+/// [`Client`] instance.
```

Comment:
```suggestion
/// The result of an attempt to receive a [`ClientRequest`] sent by a [`Client`] instance.
/// The outbound peer that would receive this request is mocked for testing.
```

Reply:
Done, I tried to phrase it to be clearer. Let me know what you think.